### ISSUE TITLE
Accordion component breaks when list of items changes

### ISSUE THREAD

**Issue by zion** [acoustep/ember-cli-foundation-6-sass#63]
Im having an issue with a list of items returned from the server after a passed in search string. As the search keyword changes the, accordion items change to properly reflect the matching results, however after the accordion `<li>`s change they no longer open/close.

Is there a way to re-initiate foundation on the accordion object after the contents change?

**Comment by donaldwasserman** [comment]
Can you provide a little code to see how you're doing this?

**Comment by zion** [comment]
Figured out how to address it by adding this to the component for the `<li>` elements in the accordion. Although JShint says Foundation is not defined. oh well...

// component.js
```  didInsertElement(){
    new Foundation.Accordion(this.$().parent(), { multiExpand: true, allowAllClosed: true});
  },```

**Comment by obernal** [comment]
@donaldwasserman I had the same problem while working with `zf-accordion` and adding elements to it after the first render. The problem has to do with `didInsertElement` being called only on the initial render as per the docs https://guides.emberjs.com/v2.13.0/components/the-component-lifecycle/. I was able to resolve this by changing `didInsertElement` on the `zf-widget` mixin to `willRender`. After that, everything worked well. That might not be the best place to put it or the best way to deal with the problem. I basically just looked for a hook that was called on every render and tested to see if it worked :). Let me know if you need a PR and I'll submit it.

**Comment by GCheung55** [comment]
I haven't looked into this too deeply but at first glance it sounds like `didReceiveAttrs` or an observer would be better suited.

**Comment by zion** [comment]
@obernal Did you override the mixin then?

**Comment by obernal** [comment]
@zion yes. My items are not components and I think it should be the accordion's responsibility to accept new items anyway so that's why this would require changes to the actual code. @GCheung55 `didReceiveAttrs` is called on re-render as well but I believe is called if the component's attributes change only. We are not changing attributes, but adding list items so I'm not sure it'll work. I will try it, but is there a reason `willRender` is not a good place for it? I haven't worked with ember much and I'm not 100% familiar with the code the mixin is executing so I'm mostly just curious.

**Comment by zion** [comment]
Ok, I seem to have broken all of my accordions by updating to ember-cli 2.13.2 and foundation 6.3.1 

I might try your solution to see if that helps
